sdk/src/client/api/block_builder/input_selection/mod.rs

```diff
@@ -38,16 +41,27 @@ pub struct InputSelection {
     required_inputs: HashSet<OutputId>,
     forbidden_inputs: HashSet<OutputId>,
     selected_inputs: Vec<InputSigningData>,
+    context_inputs: HashSet<ContextInput>,
     outputs: Vec<Output>,
     addresses: HashSet<Address>,
     burn: Option<Burn>,
     remainder_address: Option<Address>,
-    protocol_parameters: ProtocolParameters,
-    slot_index: SlotIndex,
+    slot_commitment_id: SlotCommitmentId,
     requirements: Vec<Requirement>,
     automatically_transitioned: HashSet<ChainId>,
-    mana_allotments: u64,
+    auto_mana_allotment: Option<AutoManaAllotment>,
+    mana_allotments: BTreeMap<AccountId, u64>,
     mana_rewards: HashMap<OutputId, u64>,
+    payload: Option<TaggedDataPayload>,
+    allow_additional_input_selection: bool,
+    protocol_parameters: ProtocolParameters,
+}
+
+/// Account and RMC for automatic mana allotment
+#[derive(Copy, Clone, Debug)]
+pub struct AutoManaAllotment {
+    issuer_id: AccountId,
+    reference_mana_cost: u64,
```

Comment:
Derive Getters?

Reply:
why?

Reply:
Just thought because it's a public struct, but maybe better to make it just pub(crate)?

Reply:
okay, that makes sense